sklearn/tree/_criterion.pyx

```diff
@@ -445,9 +579,12 @@ cdef class ClassificationCriterion(Criterion):
                 self.weighted_n_left += w
 
         else:
+            # Reverse reset the available values alone.
+            direction = self.missing_direction
             self.reverse_reset()
+            self.move_missing(direction)
```

Comment:
No no no, I dont think this is correct. Why are you changing the direction of the missing values here? The if-statement above is only a trick to reduce the number of operations. 

I know that you are busy as well, but could you confirm @arjoly ?